src/debugpy/_vendored/pydevd/setup_cython.py

```diff
@@ -161,12 +161,43 @@ def build_extension(dir_name, extension_name, target_pydevd_name, force_cython,
 
         # Always compile the .c (and not the .pyx) file (which we should keep up-to-date by running build_tools/build.py).
         from distutils.extension import Extension
-        ext_modules = [Extension("%s%s.%s" % (dir_name, "_ext" if extended else "", target_pydevd_name,),
-                                 c_files,
-                                 # uncomment to generate pdbs for visual studio.
-                                 # extra_compile_args=["-Zi", "/Od"],
-                                 # extra_link_args=["-debug"],
-                                 )]
+        extra_compile_args = []
+        extra_link_args = []
+
+        if 'linux' in sys.platform:
+            # Enabling -flto brings executable from 4MB to 0.56MB and -Os to 0.41MB
+            # Profiling shows an execution around 3-5% slower with -Os vs -O3,
+            # so, kept only -flto.
```

Comment:
Note: if the additional reduction from `-O3` to `-Os` makes a difference in building for more versions out of the box (150kb/.so), the 3-5% performance penalty may be reasonable.

Note that I did the checks here with `gcc 7.5.0`.